src/nl/hannahsten/texifyidea/intentions/LatexAddLabelIntention.kt

```diff
@@ -113,4 +82,27 @@ open class LatexAddLabelIntention(val command: SmartPsiElementPointer<LatexComma
 
         return candidate
     }
+
+    data class LabelWithPrefix(val prefix: String, val base: String) {
+        val prefixText = "$prefix:"
+        val labelText = "$prefix:$base"
+    }
+
+    protected fun createLabelAndStartRename(
+        editor: Editor,
+        project: Project,
+        command: LatexCommandWithParams,
+        label: LabelWithPrefix,
+        moveCaretTo: RangeMarker
+    ) {
+        val helper = LatexPsiHelper(project)
+        val parameter = helper.setOptionalParameter(command, "label", "{${label.labelText}}")
+        PsiDocumentManager.getInstance(project).doPostponedOperationsAndUnblockDocument(editor.document)
+        val parameterText =
+            parameter.keyvalValue!!.keyvalContentList.first().parameterGroup!!.parameterGroupText!!.parameterTextList.first()
```

Comment:
No safe calls. => `?.` and `firstOrNull()`. I don't know exactly if these values can be null, if that's not expected add a specific error message.

Reply:
I think, unless there is a bug in `setOptionalParameter`, the values cannot be null. `setOptionalParameter` should ensure the appropriate structure. If you want I can still use safe calls as a second second safety net.

Reply:
It's fine by me either way, `!!` should seldomly be used hence why I commented it. It's virtually always better to just go the extra mile and just handle it.

I've seen too many 'interesting' results in the psi tree the past years that I don't inherently trust it ^^

Reply:
I share your suspicion :D. How about a check and an AssertionError if the assumption is violated? This way the assumption would be explicitely expressed and documented.

Reply:
Sure. You can check for it, or do a quick `check ?: error("message")`. Do whatever you think fits best :)